src/server.rs

```diff
@@ -3,19 +3,28 @@ use std::process;
 
 use anyhow::{Context, Result};
 use axum::headers::{Header, HeaderName, UserAgent};
-use axum::{extract, response::IntoResponse, Extension, Router, Server, TypedHeader};
+use axum::{extract, response::IntoResponse, Extension, Json, Router, Server, TypedHeader};
 use axum::{routing::get, routing::put};
 use http::{HeaderValue, StatusCode};
 use tokio::sync::mpsc;
 
 use crate::model;
-use crate::settings::ServerSettings;
-pub async fn run_server(settings: ServerSettings, tx: mpsc::Sender<model::Report>) -> Result<()> {
+use crate::settings::{DBSettings, ServerSettings};
+
+pub async fn run_server(
+    settings: ServerSettings,
+    db_settings: DBSettings,
+    tx: mpsc::Sender<model::Report>,
+) -> Result<()> {
     Server::bind(&settings.host.parse::<SocketAddr>()?)
         .serve(
             Router::new()
                 .route("/health", get(health_check))
                 .route("/report-usage-stats/push", put(save_report))
+                .route(
+                    "/aggregated-stats/:day",
+                    get(|day| get_aggregated_stats(db_settings, day)),
```

Comment:
No need to manually add state to the functions. You can use axum state or axum extensions.
Also, the DB settings are just the connection settings, you most likely want to pass the DB pool already or it would connect every time again. Or how do the other endpoints do it?

Reply:
>  You can use axum state or axum extensions.

That's what I tried to do initially, but the rest of the code is written in such a way that the db access is isolated in `database.rs`. Also, functions in `database.rs` use [sorta like lazy static pattern](https://github.com/famedly/barad-dur/blob/25d36c2aeb2f7ae7a0df6d50e8fef011e56f13a0/src/database.rs#L73) which I was not sure I should touch. So that's why I do not pass pool to the endpoints handler. Other endpoints do not interact with db at all, only special threads and on of those threads is completely isolated (aggregate loop), the other is recieving data from channel. 

The other point why I abandoned state idea is because `save_report` (which uses `extract::Extension` would have a parameter (db settings or db pool) that it doesn't need, which would make [mocking it](https://github.com/famedly/barad-dur/blob/25d36c2aeb2f7ae7a0df6d50e8fef011e56f13a0/src/server.rs#L106) a bit problematic. It would be nice to solve it with something we recently discussed (`HasField` traits or smth), but currently rust doesn't have that. Another way would be to put some lenses when we construct a router, like:
```rust
  .route("/report-usage-stats/push", |s: Extension<State>, ...| put(save_report(s.only_things_related_to_save_report, ...))
```
But my first attempt resulted in some "temporary value created" or smth error, so I just did the easiest, but not prettiest thing.

Or we can start another db thread and use channels (or rather start tower service because we need request response here).

What do you think?

Reply:
If this works, I think it is good for now. We have plans to significantly refactor barad-dur at some point relatively soon as bumping f.x. axum will require some parts to be reworked. I think that would be a good opportunity to reevaluate the structure of the existing code as well, but let's go with this for now.